strings/base_coroutine_foundation.h

```diff
@@ -801,4 +801,43 @@ WINRT_EXPORT namespace winrt
         impl::check_status_canceled(shared->status);
         co_return shared->result.GetResults();
     }
+
+    template <typename Delegate>
+    struct async_event : impl::event_base<Delegate>
+    {
+        async_event() = default;
+
+        template<typename...Arg>
+        Windows::Foundation::IAsyncAction operator()(Arg const... args)
+        {
+            auto targets = this->make_temp_targets();
+
+            if (!targets)
+            {
+                co_return;
+            }
+
+            std::vector<Windows::Foundation::IAsyncAction> workers;
+            workers.reserve(targets->size());
+
+            for (auto&& element : *targets)
+            {
+                workers.push_back(invoke(element, args...));
+            }
+
+            for (auto&& element : workers)
+            {
+                co_await element;
+            }
+        }
+
+    private:
+
+        template <typename... Args>
+        Windows::Foundation::IAsyncAction invoke(Delegate const& element, Args const&... args)
+        {
+            co_await resume_background();
```

Comment:
I still think you need to let the user of the async_event determine whether there's per-delegate threads launched, or if all the delegates are fired on a single thread. My recollection is that WRL's async-event does all-on-single-new-thread, so someone coming from that space might be surprised when there's delegates competing on calling back during their operations.

Reply:
This can be achieved with the existing synchronous `winrt::event` - it executes everything in series but is fully thread-safe so you can simply call it in parallel.

Reply:
So for "execute in series on background thread", the pattern is
```cpp
auto lifetime = get_strong(); // prevent premature destruction
co_await resume_background();
m_event.invoke(*this, args);
```
?

Reply:
Yes, that would work.

Reply:
so glad to see this. we can finally stop using wrlasyncevent.h

Reply:
If you guys are happy with "execute in series on background thread" then I can close this PR and avoid introducing a new event type.

Reply:
WRL-based AsyncEvent raised all events serialized on a single background thread, so I don't think we need the parallel invoke.

Reply:
Then my work here is done. 😉